### DISCUSSION TITLE
PowerShell and .NET version number alignment ?

### DISCUSSION THREAD

**Discussion by fMichaleczek** [PowerShell/PowerShell#19982] [Polls]
There is the possibility of aligning the version numbers of PowerShell and .NET on next November with the next Stable release of .NET and PowerShell.

However this is not the direction the PowerShell team has chosen.

I do not understand this choice, so I launch this discussion and this vote to see the opinion of the community.

**Comment by JustinGrote** [comment]
Depends on how we think of PowerShell. If we think of it as integral to dotnet as something like ASP.NET, then it should follow similar versioning. Since PowerShell is pinned to the .NET lifecycle, it would make sense to align things so that similar messaging about support lifecycle, etc. can be communicated simply via the version number.

If we see it as its own standalone application, then reserving major version changes for large breaking changes or indicating major improvements seems reasonable outside the .NET ecosystem.

As I recall the PS team views it as the latter.

**Reply by fMichaleczek** [reply]
Thanks for the clarity !

Switching to dotnet versioning is also assumed that the version will be number 15 in 2030.
Did those who voted yes understand this consequence?

I think those more IT or committed to security have another more pragmatic approach (make me a bunch with the same number please).

**Comment by bergmeister** [comment]
Agree with Justin, after so many iterations of .NET, the new additions aren't that relevant any more to the majority of PS users who do not use webserver features, it's mostly just perf improvements.
Given the coupled release cycle, I wouldn't mind aligning the minor version number to it. But the confusion of skipping a few numbers is probably not worth it when it would be just convenient but is actually not a problem that needs to be solved.

**Reply by JustinGrote** [reply]
7.8 aligning to .NET 8 is a good idea, I like it. Future major changes would then skip to 8.8 for example.

This is making the big assumption .NET doesn't change their versioning scheme again, it's been about 5 years so they are about due :)
